src/lib/uname.ts

```diff
@@ -0,0 +1,72 @@
+import { TypedError } from 'typed-error';
+import { exec } from './fs-utils';
+import log from './supervisor-console';
+
+export class UnameError extends TypedError {}
+
+// Exported for tests
+export async function get(flags = ''): Promise<string> {
+	const { stdout, stderr } = await exec(`uname${flags ? ` ${flags}` : ''}`);
+	if (stderr.length > 0) {
+		throw new UnameError(`Error querying uname: ${stderr}`);
+	}
+	return stdout.toString().trim();
+}
+
+export async function getKernelSlug(): Promise<string | undefined> {
+	try {
+		// `exports` makes `get` mockable for testing.
+		const uname = await exports.get('-s');
```

Comment:
This a code smell, tests should not rely on mocks as much as possible because that means the test needs to know internal implementation of the function, which makes the test brittle to implementation changes. You should always try to use dependency injection in a case like this.

In this case that means that the `getProp` functions receive as argument a function that returns `uname` or you split the implementation into `parseKernelVersion()`, `parseL4TVersion()` that receive the output of uname and test those. Then the `getProp` functions are trivial and don't need testing. You just need to call uname and pass the result to the corresponding `parseProp` function